Source/JavaScriptCore/runtime/StringPrototype.cpp

```diff
@@ -706,7 +706,7 @@ JSC_DEFINE_HOST_FUNCTION(stringProtoFuncSplitFast, (JSGlobalObject* globalObject
 
         unsigned resultSize = result.size();
         if (limit == 0xFFFFFFFFu && !globalObject->isHavingABadTime() && resultSize < MIN_SPARSE_ARRAY_INDEX) [[likely]] {
-            bool makeAtomStringsArray = resultSize < atomStringsArrayLimit;
+            bool makeAtomStringsArray = false; // temporarily disabled because of a performance regression, see PR comment for details
```

Comment:
We should make this comment: 
```
// FIXME: Temporarily disabled because of a performance regression. See: <link to bugzilla to fix it with explanation of what went wrong>
```

That way folks don't have to dig through the commit history. It also makes it harder to forget to file a bug to actually fix this :)